extra/models/inception.py

```diff
@@ -0,0 +1,308 @@
+    self.Mixed_7a = get_cls("D1","D",InceptionD)(768)
+    self.Mixed_7b = get_cls("E1","E",InceptionE)(1280)
+    self.Mixed_7c = get_cls("E2","E",InceptionE)(2048)
+    self.avgpool = lambda x: Tensor.avg_pool2d(x, kernel_size=(8,8), padding=1)
+    self.fc = Linear(2048, num_classes)
+
+  def __call__(self, x:Tensor) -> Tensor:
+    return x.sequential([
+      self.Conv2d_1a_3x3,
+      self.Conv2d_2a_3x3,
+      self.Conv2d_2b_3x3,
+      self.maxpool1,
+
+      self.Conv2d_3b_1x1,
+      self.Conv2d_4a_3x3,
+      self.maxpool2,
+
+      self.Mixed_5b,
+      self.Mixed_5c,
+      self.Mixed_5d,
+      self.Mixed_6a,
+      self.Mixed_6b,
+      self.Mixed_6c,
+      self.Mixed_6d,
+      self.Mixed_6e,
+
+      self.Mixed_7a,
+      self.Mixed_7b,
+      self.Mixed_7c,
+      self.avgpool,
+
+      lambda y: y.reshape(x.shape[0],-1),
+      self.fc,
+    ])
+
+
+# FID Inception Variation
+
+class FidInceptionA(InceptionA):
+  def __call__(self, x:Tensor) -> Tensor:
+    outputs = [
+      self.branch1x1(x),
+      x.sequential([self.branch5x5_1, self.branch5x5_2]),
+      x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2, self.branch3x3dbl_3]),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False))
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionC(InceptionC):
+  def __call__(self, x:Tensor) -> Tensor:
+    outputs = [
+      self.branch1x1(x),
+      x.sequential([self.branch7x7_1, self.branch7x7_2, self.branch7x7_3]),
+      x.sequential([self.branch7x7dbl_1, self.branch7x7dbl_2, self.branch7x7dbl_3, self.branch7x7dbl_4, self.branch7x7dbl_5]),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False))
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionE1(InceptionE):
+  def __call__(self, x:Tensor) -> Tensor:
+    branch3x3 = self.branch3x3_1(x)
+    branch3x3dbl = x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2])
+    outputs = [
+      self.branch1x1(x),
+      Tensor.cat(self.branch3x3_2a(branch3x3), self.branch3x3_2b(branch3x3), dim=1),
+      Tensor.cat(self.branch3x3dbl_3a(branch3x3dbl), self.branch3x3dbl_3b(branch3x3dbl), dim=1),
+      self.branch_pool(x.avg_pool2d(kernel_size=(3,3), stride=1, padding=1, count_include_pad=False)),
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+class FidInceptionE2(InceptionE):
+  def __call__(self, x:Tensor) -> Tensor:
+    branch3x3 = self.branch3x3_1(x)
+    branch3x3dbl = x.sequential([self.branch3x3dbl_1, self.branch3x3dbl_2])
+    outputs = [
+      self.branch1x1(x),
+      Tensor.cat(self.branch3x3_2a(branch3x3), self.branch3x3_2b(branch3x3), dim=1),
+      Tensor.cat(self.branch3x3dbl_3a(branch3x3dbl), self.branch3x3dbl_3b(branch3x3dbl), dim=1),
+      self.branch_pool(x.max_pool2d(kernel_size=(3,3), stride=1, padding=1)),
+    ]
+    return Tensor.cat(*outputs, dim=1)
+
+@lru_cache()
```

Comment:
i would not lru_cache this, it won't be free after it's loaded. also this can just inline in the load function

otherwise looks good, will merge after this is fixed